client/src/Api.js

```diff
@@ -65,4 +65,69 @@ export async function deleteProfilePicture(token) {
     .catch((error) => {
       throw error;
     });
-}
\ No newline at end of file
+}
+export async function fetchPosts({ userId = null, page = 1, limit = 10 } = {}) {
+  await servicesWait();
+
+  const params = {
+    page,
+    limit,
+    ...(userId && { user_id: userId }), 
+  };
+
+  return axios
+    .get(`${services.controller.url}/post/posts`, { params })
+    .then((response) => response.data.posts)
+    .catch((error) => {
+      throw error;
+    });
+}
+
+
+export async function addPost(token, description, images, location = "") {
+  const formData = new FormData();
+  formData.append("description", description);
+  if (location) formData.append("location", location);
+
+  images.forEach((image, index) => {
+    formData.append("images", image); 
+  });
+
+  return axios
+    .post(`${services.controller.url}/post/posts`, formData, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "multipart/form-data",
+      },
+    })
+    .then((response) => response.data.post)
+    .catch((error) => {
+      throw error;
+    });
+}
+
+export const deletePost = async (token, postId) => {
```

Comment:
Git, ale możesz napisać export async function deletePost(token, postId) {
To powinno być chyba to samo, chyba że jest jakiś powód dla którego to jest tak zrobione

Reply:
a rzeczywiscie, tu sie bawilam bo mi cos nie dzialalo ale to w czyms innym byl problem, poprawie